base/tuple.jl

```diff
@@ -32,6 +32,11 @@ getindex(t::Tuple, r::AbstractArray{<:Any,1}) = (eltype(t)[t[ri] for ri in r]...
 getindex(t::Tuple, b::AbstractArray{Bool,1}) = length(b) == length(t) ? getindex(t, findall(b)) : throw(BoundsError(t, b))
 getindex(t::Tuple, c::Colon) = t
 
+# get methods (getindex with default value) (Issue #40809)
+
+get(t::Tuple, i::Int, default) = i <= length(t) ? getindex(t, i) : default
+get(f::Callable, t::Tuple, i::Int) = i <= length(t) ? getindex(t, i) : f()
```

Comment:
I think these should allow `get((1, 2), 0, 99)` to work, since `get((a=1, b=2), 0, 99)` does. And `(1,2)[Int32(1)]` does too.
```suggestion
get(t::Tuple, i::Integer, default) = i in 1:length(t) ? getindex(t, i) : default
get(f::Callable, t::Tuple, i::Integer) = i in 1:length(t) ? getindex(t, i) : f()
```

Reply:
Hi @mcabbott , could you please help me understand why you replaced `<=` with a range?

Reply:
> was that a short way of eliminating negative indices?

I would think so.